### PR TITLE
 Replace avatar solution with react-avatar

### PR DESCRIPTION
**Pull request by galileo**
**Release Type:** *Non-Breaking Feature* 

Fixes https://x-team-internal.atlassian.net/browse/XP-2681

## Description

To be able to render an avatar with **XP** value we decide to change the way we are creating avatars.
On top of that we added the **react-avatar** library.

One additional thing to mention is that there was a change introduce with from the note.

As we were adding a author_name to the **alt** property for the Avatar, and the decision was to name the create new notes with the author name value as **Remote experience details (legacy field)** I was forced to add a change into this value. That we will only add an alt name from the author name if the avatar will not be set up. In any other case I would need to add a new **Notes type** or add an if to handle the special case for the author_avatar field.

We also agreed that we don't want to spent much time to solve the issue with spread avatar on ApplicantBadge tabular view.

## Checklist

**Before submitting a pull request,** please make sure the following is done:

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `IN REVIEW` column
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`)
- [x] check cleanup tasks (https://github.com/x-team/auto/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases or variants usage
- [x] if you've fixed a bug, make sure to also include a new story that will expose declarations with problematic data that caused the bug in the first place, so that we can assure that no regressions will pop up on future interactions
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot
- [x] **manually tested the component** by running it in the browser and checked nothing is broken and operates as expected!

## Related PRs

branch | PR
------ | ------
XP-2681-migrate-remote-experience-details | [Auto](https://github.com/x-team/auto/pull/1791)
XP-2681-migrate-remote-experience-details | [Auto API](https://github.com/x-team/auto-api/pull/614)

## Steps to Test or Reproduce

Check current avatar sections.

## Screenshots
avatar

### CODE REVIEW

**Comment by nicksp** [comment]
@galileo I believe this PR is ready for review, right? So I set a proper label to it 😏

File: src/components/ui/Avatar.js [hunk 1]
```diff
@@ -1,47 +1,40 @@
...
-      })
-    )
-  }
+const Avatar = (props: Props) => {
+  const { alt, ...restProps } = props
+
+  return <ReactAvatar
+    className={cx.avatarSizeConstraint}
+    name={alt}
+    color={`${theme.baseRed}`}
```
[the remark this answers is not in the document]
Comment (galileo) [reply src/components/ui/Avatar.js]: This can not be done because the color is an object and the type for the **color property** is string so I need have a way to stringify the object. I found this the cleaniest way to do. 

This is how the console log for this the theme.baseRed is looking.
zrzut ekranu z 2019-01-31 07-30-03

**Comment by galileo** [comment]
The case with Batman url is a little bit complicated to solve. I mean we need to decide what to do. Because 
react-avatar is expecting the sources to be a valid image files. This mean the Content-Type for the image should be set correct accordingly to it's type. 
The batman image doesn't have the content-type so the library is not threating this as valid image resource.
Also there is a policy CORB policy applied. You can watch the policy in here.
https://chromium.googlesource.com/chromium/src/+/master/services/network/cross_origin_read_blocking_explainer.md

And the issue for react-avatar in here:
https://github.com/Sitebase/react-avatar/issues/137

So, in other words, if the resource will be a valid image resource the react-avatar will handle it, if not it will display initials. 

Let me know how to proceed because this is not something that I can change. We can agree if this is fine for us and change the example image for an valid image resource or we will be forced to uninstall the react-avatar library. If you have any additional idea please let me know.

**Comment by galileo** [comment]
Thank you @nicksp for the review.

The case with Batman url is a little bit complicated to solve. I mean we need to decide what to do. Because 
react-avatar is expecting the sources to be a valid image files. This mean the Content-Type for the image should be set correct accordingly to it's type. 
The batman image doesn't have the content-type so the library is not threating this as valid image resource.

Also there is a policy CORB policy applied. You can watch the policy in here.

https://chromium.googlesource.com/chromium/src/+/master/services/network/cross_origin_read_blocking_explainer.md

And the issue for react-avatar in here:
https://github.com/Sitebase/react-avatar/issues/137

So, in other words, if the resource will be a valid image resource the react-avatar will handle it, if not it will display initials. 

Let me know how to proceed because this is not something that I can change. We can agree if this is fine for us and change the example image for an valid image resource or we will be forced to uninstall the react-avatar library. If you have any additional idea please let me know.

File: src/components/ui/Avatar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicksp) [reply src/components/ui/Avatar.js]: Ah right, that's because `color` prop expects a string, makes sense. What I usually do (and it's done like that in our codebase) in such cases is this, which is more explicit: `color={theme.baseRed.toString()}`. And there's no need to interpolate 😏 So we can adjust to be like that also

**Comment by nicksp** [comment]
> And the issue for react-avatar in here:
Sitebase/react-avatar#137

Ah I see what the issue is, weird one. Alright, this isn't an issue for us, as I can see throughut me testing the app. Avatarts for applicants are loaded correctly. But for stories case let's change images so we can show that we can render them, otherwise stories for `Avatar` don't cover that case. So, please find any suitable pictire and use instead of Batman in existing stories that I mentioned in my previous review. Sounds OK?

File: src/components/ui/Avatar.js [hunk 2]
```diff
@@ -1,47 +1,40 @@
...
+    maxInitials={3}
+    round
+    {...restProps}
```
[the remark this answers is not in the document]
Comment (nicksp) [reply src/components/ui/Avatar.js]: @galileo So, are you going to apply that? Or did I miss the commit?

[the remark this answers is not in the document]
Comment (galileo) [reply src/components/ui/Avatar.js]: Yes. I applied it.
